File: qif4j/src/main/java/com/github/squirrelgrip/qif4j/write/QifInvestmentAction.java
package com.github.squirrelgrip.qif4j.write;

public enum QifInvestmentAction {
	BUY("Buy"), SELL("Sell"), DIV("Div"), INTEREST_INCOME("IntInc"), TRANSFER_IN("XIn"), TRANSFER_OUT("XOut");
	
	private String text;
	
	private QifInvestmentAction(String text) {
		this.text = text;
	}
	
	public String getText() {
		return text;
	}

	public static QifInvestmentAction forText(String text) {
		for(QifInvestmentAction action: QifInvestmentAction.values()) {
			if(action.getText().equals(text)) {
				return action;
			}
		}
		throw new IllegalArgumentException(text + " is not a QIF Invst action");
	}
}

File: qif4k/src/main/kotlin/com/github/squirrelgrip/qif4k/write/QifInvestmentAction.kt
package com.github.squirrelgrip.qif4k.write

enum class QifInvestmentAction(val text: String) {
    BUY("Buy"), SELL("Sell"), DIV("Div"), INTEREST_INCOME("IntInc"), TRANSFER_IN("XIn"), TRANSFER_OUT("XOut");

    companion object {
        fun forText(text: String?): QifInvestmentAction {
            for (action in values()) {
                if (action.text == text) {
                    return action
                }
            }
            throw IllegalArgumentException("$text is not a QIF Invst action")
        }
    }

}